### PR TITLE
builder/amazon: Support for source instance tenancy

### PR DESCRIPTION
**Pull request by epipho**
Allow for packer to spin up source instances using dedicated hosts by setting the [Tenancy](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/dedicated-instance.html) value.

### CODE REVIEW

**Comment by epipho** [comment]
It looks like check-generate is failing on some unrelated files. These are different on my machine as well, looks like newline differences. Would you like me to to add the files to this PR as well?

**Comment by SwampDragons** [comment]
Strange; I can't reproduce the generate weirdness. try rebasing on the master branch?

File: builder/amazon/common/run_config.go [hunk 1]
```diff
@@ -631,6 +635,12 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		}
 	}
 
+	if c.Tenancy != "" && c.Tenancy != "default" {
+		if c.SpotPrice != "" {
+			errs = append(errs, fmt.Errorf("Error: Non-default tenancy cannot be used in conjunction with Spot Instances"))
```
Comment (SwampDragons) [inline builder/amazon/common/run_config.go:640]: Why is this the case for Packer? It looks like this is possible based on AWS docs.

**Review by SwampDragons — CHANGES_REQUESTED** [review]
One tiny request around validation in the Prepare, and I'm wondering why we can't do this for Spot instances too, but overall this looks great and behaves as expected. Nice going!

File: builder/amazon/common/run_config.go [hunk 2]
```diff
@@ -631,6 +635,12 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		}
 	}
 
+	if c.Tenancy != "" && c.Tenancy != "default" {
```
[the remark this answers is not in the document]
Comment (epipho) [reply builder/amazon/common/run_config.go]: Added validation and tests

File: builder/amazon/common/run_config.go (hunk 1 above)
Comment (epipho) [reply builder/amazon/common/run_config.go]: Indeed this looks to be the case, removed check.

**Comment by SwampDragons** [comment]
This looks good -- docs need to be regenerated before we can merge though

**Comment by epipho** [comment]
Whoops, forgot to regenerate after rebasing.  Fixed, looks like all green now.

**Review by SwampDragons — APPROVED** [review]

**Comment by SwampDragons** [comment]
thanks!

**Pull request merged**